docs/designs/016-audit-logging.md

```diff
@@ -0,0 +1,553 @@
+```
+
+### Audit Logger Package
+
+The new `auditlogger` package follows the same pattern as the existing `logger` package but writes to a file:
+
+```go
+package auditlogger
+
+import (
+    "log/slog"
+    "os"
+    "path/filepath"
+)
+
+const (
+    DefaultAuditLogDir  = "/var/log/nvsentinel"
+    EnvAuditLogPath     = "AUDIT_LOG_PATH"
+    EnvAuditEnabled     = "AUDIT_ENABLED"
+    EnvPodName          = "POD_NAME"
+)
+
+var (
+    auditLogger *slog.Logger
+    auditFile   *os.File  // Store file handle for proper cleanup
+)
+
+// InitAuditLogger initializes the audit logger for a component
+func InitAuditLogger(component string) error {
+    if os.Getenv(EnvAuditEnabled) == "false" {
+        return nil  // Audit logging disabled
+    }
+    
+    // Build log path: /var/log/nvsentinel/{POD_NAME}-audit.log
+    logPath := os.Getenv(EnvAuditLogPath)
+    if logPath == "" {
+        podName := os.Getenv(EnvPodName)
+        if podName == "" {
+            podName = component  // Fallback to component name
+        }
+        logPath = filepath.Join(DefaultAuditLogDir, podName+"-audit.log")
+    }
+    
+    // Ensure directory exists
+    if err := os.MkdirAll(filepath.Dir(logPath), 0755); err != nil {
+        return err
+    }
+    
+    // Open file for append
+    var err error
+    auditFile, err = os.OpenFile(logPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+    if err != nil {
+        return err
+    }
+    
+    // Create JSON handler writing to file
+    auditLogger = slog.New(slog.NewJSONHandler(auditFile, &slog.HandlerOptions{}))
+    auditLogger = auditLogger.With("component", component)
+    
+    return nil
+}
+
+// CloseAuditLogger flushes and closes the audit log file
+// Should be called during graceful shutdown (SIGTERM handler)
+func CloseAuditLogger() error {
+    if auditFile != nil {
+        return auditFile.Close()
+    }
+    return nil
+}
+
+// Log writes a minimal audit entry
+func Log(method, url string) {
+    if auditLogger == nil {
+        return  // Audit logging not initialized or disabled
+    }
+    
+    auditLogger.Info("audit",
+        "method", method,
+        "url", url,
+    )
+}
+```
+
+### Log Rotation
+
+NVSentinel uses [lumberjack](https://github.com/natefinch/lumberjack) for built-in log rotation:
+
+```go
+import (
+    "log/slog"
+    "gopkg.in/natefinch/lumberjack.v2"
+)
+
+// Create lumberjack logger with rotation
+lumberjackLogger := &lumberjack.Logger{
+    Filename:   logPath,
+    MaxSize:    100,  // megabytes
+    MaxBackups: 7,    // number of backups
+    MaxAge:     30,   // days
+    Compress:   true, // compress rotated files
+}
+
+// Use lumberjack as slog output
+auditLogger = slog.New(slog.NewJSONHandler(lumberjackLogger, nil))
+```
+
+**Output format (verified via POC):**
+```json
+{"time":"2025-11-27T19:29:26Z","level":"INFO","msg":"audit","component":"fault-quarantine","method":"PATCH","url":"/api/v1/nodes/gpu-node-01"}
```

Comment:
The struct above says timestamp, but here we have time. Can we pick on option and stick to it for consistency?